buddy_list: Change how "Invite users to organization" link is shown.

evykassirer opened the pull request:
Fixes #31653

**part (1)**

> Add a three-dot menu to the right of the search button, with the "Invite users to organization" link as the only menu item to start with. We should use the same icon as we have in the gear menu. If a user doesn't have permission to send invites, the three-dot menu button should not be shown.

![Kapture 2024-09-30 at 16 19 48](https://github.com/user-attachments/assets/f8dc888e-700f-4b54-ba4d-eb493e20ef19)

guest has no three-dot menu:

<img width="261" alt="image" src="https://github.com/user-attachments/assets/daac611f-051d-4e69-af7d-fdff9a84e546">


**part (2)**

> Make the "Invite users to organization" link be scrollable, rather than pinning it to the bottom of the sidebar.

![Kapture 2024-09-30 at 16 21 17](https://github.com/user-attachments/assets/af04a23b-0b21-4eff-97e0-694ebdecc4d6)

evykassirer commented:
They each fixed half of the issue, so I wrote "part of fixing", but I guess I'll change the second commit to "Fixes"

karlstolley commented:
Looks good in outline form! I'd think the final commit should be to remove everything related to the blue invite link at the bottom of the right sidebar.

evykassirer commented:
The issue said "Make the "Invite users to organization" link be scrollable, rather than pinning it to the bottom of the sidebar." -- did something change to have us want to remove it?

evykassirer commented:
@timabbott (or @alya) ready for you!

karlstolley commented:
@evykassirer said:

> The issue said "Make the "Invite users to organization" link be scrollable, rather than pinning it to the bottom of the sidebar." -- did something change to have us want to remove it?

Ah, you're right. I misremembered [the CZO conversation](https://chat.zulip.org/#narrow/stream/137-feedback/topic/hiding.20inactive.20users.20in.20right.20sidebar/near/1944275) about this.

alya commented:
Thanks! Let's make they styling match the redesign better, though.

1. The styling should match menu options in the left sidebar menus.
2. As noted in the issue, we should use the same icon as we have in the gear menu. 
3. The hover behavior should match the design and the left sidebar as well. (The search button could get the same treatment, but I imagine we'll be converting it to an always-open search bar soon, so either way.)


![Screenshot 2024-10-01 at 13 26 43@2x](https://github.com/user-attachments/assets/7748b112-f54a-43db-b722-5aba83c5f518)

![Screenshot 2024-10-01 at 13 27 49@2x](https://github.com/user-attachments/assets/e688db32-e1ab-464c-8ac9-9422f7a5438a)

karlstolley commented on web/templates/right_sidebar.hbs:
```diff
@@ -2,10 +2,12 @@
+                    <h4 class='right-sidebar-title' id='userlist-title'>
+                        {{t 'USERS' }}
+                    </h4>
+                    <i id="user_filter_icon" class="fa fa-search"></i>
```
Should we change this to `.zulip-icon-search` here?

evykassirer replied:
There's a different design for search, so I think it makes sense to just not touch it for now.

karlstolley commented on web/styles/right_sidebar.css:
```diff
@@ -364,6 +364,15 @@ $user_status_emoji_width: 24px;
 
+    #buddy-list-menu-icon {
+        opacity: 0.5;
+        justify-self: right;
```
The `justify-self` here isn't necessary, as the `auto` column definition is just going to let the icon take up what space it will in the grid anyway.

evykassirer replied:
Removed, thanks

karlstolley commented:
Although the old button now scrolls, it seems like we should think about the block of negative, empty space below the user list:

![Screenshot 2024-10-02 at 12 42 38 PM](https://github.com/user-attachments/assets/62835f9d-d48a-439f-b48c-f7318cdeed84)

evykassirer commented on web/styles/right_sidebar.css:
```diff
@@ -339,10 +339,17 @@ $user_status_emoji_width: 24px;
     margin-right: var(--width-simplebar-scroll-hover);
 
+    #userlist-header-search {
+        display: grid;
```
Ah yeah I did it because of the click/hover targets, but I can do the "multiple selectors to same click/hover event" route instead.

evykassirer replied:
I'm a bit worried about `user_filter_icon` not having that much margin/padding on it. If I did multiple selectors, the click (to search) and hover (for tooltip) target would change from this:

<img width="234" alt="image" src="https://github.com/user-attachments/assets/1a64c235-dca5-4fb4-9ab9-34b905183ebc">

to this:

<img width="222" alt="image" src="https://github.com/user-attachments/assets/86199e66-dc39-468c-b378-19b858c50067">

Which I don't super like, so I'm going to leave it as it is currently until we talk about it more.

evykassirer replied:
@karlstolley just following up on this!

karlstolley replied:
Yeah. The biggest thing is not having the deadspace between the icon and the heading. But as that heading might be going away anyway, the solution you came up with initially is probably best for multiple reasons.

evykassirer commented:
Great call on the bottom empty space, I found the code that was creating it and deleted it.

evykassirer commented:
re: styling issues Alya brought up:

(1) The styling should match menu options in the left sidebar menus.

I updated the popover list to use the same style classnames as the left sidebar. I'm not sure what specifically you were seeing, but hopefully they look more similar to you now. I'm not noticing any differences.

(2) As noted in the issue, we should use the same icon as we have in the gear menu.

I updated the invite users icon to match the icon we use for that in the gear menu.

<img width="292" alt="image" src="https://github.com/user-attachments/assets/93a50bb0-d934-4419-8d74-2060ae73e423">

(3) The hover behavior should match the design and the left sidebar as well.

<img width="80" alt="image" src="https://github.com/user-attachments/assets/9609b24d-f770-4dbb-8fe8-66420174907a">


> (The search button could get the same treatment, but I imagine we'll be converting it to an always-open search bar soon, so either way.)

I agree it makes sense to just wait on that.

evykassirer commented:
(I've now addressed or replied to all comments)

alya commented:
Hmm, why am I seeing this (all the `...`s showing up)?

![Screenshot 2024-10-03 at 16 56 42](https://github.com/user-attachments/assets/3386501a-297e-417e-a9b7-cec4faa058ca)

evykassirer commented:
> Hmm, why am I seeing this (all the ...s showing up)?

I messed up a hover style. Fixed it now! Sorry I didn't catch that.

evykassirer commented:
There's a puppeteer failure, I think it's not related since I only changed a bit of CSS since last time and CI was passing before.

karlstolley commented:
There was a CI flake with Puppeteer that Aman fixed over the weekend. Re-running the failed job now.

karlstolley commented:
I've marked this for CZO review, provided of course the failing CI passes.

alya commented:
It's not necessarily a blocker for merging this, but we should make keyboard navigation work for this menu. In particular, pressing "down" once you open it should select the top option.

alya commented:
(I also just discovered https://chat.zulip.org/#narrow/stream/9-issues/topic/extra.20.22down.22.20required.20to.20select.20top.20option/near/1957947 while testing this. :))

timabbott commented on web/styles/right_sidebar.css:
```diff
@@ -49,6 +49,60 @@ $user_status_emoji_width: 24px;
     }
 }
 
+.buddy-list-section li {
```
The `li` part here is probably useless and should be deleted, possibly both here and below? But that's not new in this PR.

evykassirer replied:
It's needed for the `hover` stuff, but I could move it down to the hover styles specifically if you want.

timabbott replied:
OK but couldn't it be just `.user_sidebar_entry` instead?

timabbott commented:
I'm not sure what's wrong with the keyboard navigation; it looks like focus is getting stuck on the `\vots`, so might be another click handler being overly zealous. Merging, @evykassirer can you track that follow-up?
